xskillscore/tests/test_probabilistic.py

```diff
@@ -476,3 +478,185 @@ def test_rps_perfect_values(o, category_edges):
+)
+def test_roc_returns(
+    observation,
+    forecast,
+    symmetric_edges,
+    dim,
+    return_results,
+    input,
+    chunk,
+    drop_intermediate,
+):
+    """testing keywords and inputs pass"""
+    if "Dataset" in input:
+        name = "var"
+        forecast = forecast.to_dataset(name=name)
+        observation = observation.to_dataset(name=name)
+    if chunk:
+        forecast = forecast.chunk()
+        observation = observation.chunk()
+
+    roc(
+        observation,
+        forecast,
+        symmetric_edges,
+        dim=dim,
+        drop_intermediate=drop_intermediate,
+        return_results=return_results,
+    )
+
+
+def test_roc_auc_score_random_forecast(
+    forecast_1d_long, observation_1d_long, symmetric_edges
+):
+    """Test that ROC AUC around 0.5 for random forecast."""
+    area = roc(
+        observation_1d_long,
+        forecast_1d_long,
+        symmetric_edges,
+        dim="time",
+        return_results="area",
+    )
+    assert area < 0.6
+    assert area > 0.4
+
+
+def test_roc_auc_score_perfect_forecast(forecast_1d_long, symmetric_edges):
+    """Test that ROC AUC equals 1 for perfect forecast."""
+    area = roc(
+        forecast_1d_long,
+        forecast_1d_long,
+        symmetric_edges,
+        drop_intermediate=False,
+        dim="time",
+        return_results="area",
+    )
+    assert area == 1.0
+
+
+@pytest.mark.parametrize("drop_intermediate", [False, True])
+def test_roc_auc_score_out_of_range_forecast(
+    forecast_1d_long, observation_1d_long, symmetric_edges, drop_intermediate
+):
+    """Test that ROC AUC equals 0.0 or 0.5 for out of range forecast."""
+    area = roc(
+        observation_1d_long,
+        xr.ones_like(forecast_1d_long) + 100,
+        symmetric_edges,
+        drop_intermediate=drop_intermediate,
+        dim="time",
+        return_results="area",
+    )
+    assert float(area) in [0.5, 0.0]
+
+
+@pytest.mark.parametrize("drop_intermediate", [False, True])
+def test_roc_auc_score_out_of_range_observation(
+    forecast_1d_long, observation_1d_long, symmetric_edges, drop_intermediate
+):
+    """Test that ROC AUC equals 0.0 for out of range observation."""
+    area = roc(
+        xr.ones_like(observation_1d_long) + 100,
+        forecast_1d_long,
+        symmetric_edges,
+        drop_intermediate=drop_intermediate,
+        dim="time",
+        return_results="area",
+    )
+    np.testing.assert_almost_equal(area, 0.0, decimal=2)
+
+
+@pytest.mark.parametrize("drop_intermediate", [False, True])
+def test_roc_auc_score_out_of_range_edges(
+    forecast_1d_long, observation_1d_long, symmetric_edges, drop_intermediate
+):
+    """Test that ROC AUC equals 0.5 or 0.0 for out of range edges."""
+    area = roc(
+        observation_1d_long,
+        forecast_1d_long,
+        symmetric_edges + 100,
+        drop_intermediate=drop_intermediate,
+        dim="time",
+        return_results="area",
+    )
+    assert float(area) in [0.5, 0.0]
```

Comment:
these tests are not stable locally. sometimes they produce 0, sometimes 0.5. my idea was to test some edge cases. thoughts?

Reply:
> Took a quick look at the tests you said weren't stable.
> 
> Curious how it's unstable. I get 0 for `test_roc_auc_score_out_of_range_forecast` and 0.5 for `test_roc_auc_score_out_of_range_edges`. In my head those are expected.
> 
> ```
> import xskillscore as xs
> import xarray as xr
> import numpy as np
> np.random.seed(42)
> 
> s = 100
> o_1d_long = xr.DataArray(np.random.normal(size=(s)), coords=[("time", np.arange(s))])
> f_1d_long = xr.DataArray(np.random.normal(size=(s)), coords=[("time", np.arange(s))])
> very_bad_f = xr.ones_like(f_1d_long) + 100
> 
> symmetric_edges = np.linspace(-3, 3, 11)
> 
> xs.roc(o_1d_long, very_bad_f, symmetric_edges, drop_intermediate=True, dim="time", return_results="area") # -> 0
> xs.roc(o_1d_long, very_bad_f, symmetric_edges, drop_intermediate=False, dim="time", return_results="area") # -> 0
> 
> large_symmetric_edges = symmetric_edges + 100
> 
> xs.roc(o_1d_long, f_1d_long, large_symmetric_edges, drop_intermediate=True, dim="time", return_results="area") # -> 0.5
> xs.roc(o_1d_long, f_1d_long, large_symmetric_edges, drop_intermediate=False, dim="time", return_results="area") # -> 0.5
> ```

pushed only one allowed solutions. seems to work but repeatedly doesnt work on my local machine. would be annoying when this fails unexpected in the future... maybe there is some bug in our code explaining this...